src/components/ViewRequest/Actions/ReworkRequest.tsx

```diff
@@ -24,13 +24,30 @@ const ReworkRequest = () => {
 
   const updateHandler = () => {
     if (request.data && currentStage?.previous) {
-      const newStage = currentStage.previous;
-      const eventTitle = currentStage.previousEventTitle;
-      updateStage.mutate({
+      let newData = {
         requestId,
-        newStage,
-        eventTitle,
-      });
+        newStage: "",
+        newSubStage: "",
+        eventTitle: "",
+      };
+
+      const subStage = currentStage.subStages?.find(
+        ({ key }) => key === request.data.subStage
+      );
+
+      if (subStage && subStage.previous) {
+        newData.newStage = currentStage.key;
+        newData.newSubStage = subStage.previous;
+        newData.eventTitle = subStage.previousEventTitle;
+      } else {
+        const previousStage = STAGES.find(
+          ({ key }) => key === currentStage.previous
+        );
+        newData.newStage = currentStage.previous;
+        newData.newSubStage = previousStage?.subStages?.[0].key || ""; // first substage or empty string
```

Comment:
This might be a good question to raise with them?  Would they want to go back to the first or last substage of the previous stage?  Or should a dropdown be offered to send them to a particular stage?  Also, something I thought of, they talked about the whole Title 10 office thing, and it would only be a substage if they selected that it needed to go to Title 10... So might need some logic to determine if the substage exists or not based on criteria.